NGit/NGit.Api/ApplyCommand.cs

```diff
@@ -227,9 +226,11 @@ private void Apply(FilePath f, FileHeader fh)
 			}
 			IList<string> newLines = new AList<string>(oldLines);
 		    RawText hrt = null;
+		    var containedCRLF = false;
             foreach (HunkHeader hh in fh.GetHunks())
 			{
 				var buffer = Sharpen.Runtime.GetStringForBytes(hh.GetBuffer(), hh.GetStartOffset(), hh.GetEndOffset() - hh.GetStartOffset());
+			    containedCRLF |= buffer.Contains("\r\n");
```

Comment:
We need to know if there if there are any windows line endings in the patch. so instead of doing it when parsing we do it when we read in from the buffer.